dev-embedded/sdcc/sdcc-4.2.0.ebuild

```diff
@@ -0,0 +1,167 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools
+
+if [[ ${PV} == "9999" ]] ; then
+	ESVN_REPO_URI="https://svn.code.sf.net/p/sdcc/code/trunk/sdcc"
+	inherit subversion
+else
+	SRC_URI="
+		mirror://sourceforge/sdcc/${PN}-src-${PV}.tar.bz2
+		doc? ( mirror://sourceforge/sdcc/${PN}-doc-${PV}.tar.bz2 )
+	"
+
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Small device C compiler (for various microprocessors)"
+HOMEPAGE="http://sdcc.sourceforge.net/"
+
+LICENSE="
+	GPL-2 ZLIB
+	non-free? ( MicroChip-SDCC )
+	packihx? ( public-domain )
+"
+SLOT="0"
+# in order of configure.ac's AC_DO_PORT stanzas
+SDCC_PORTS="
+	mcs51
+	z80 z180
+	r2k r2ka r3ka
+	sm83
+	tlcs90
+	ez80-z80
+	z80n
+	ds390 ds400
+	pic14 pic16
+	hc08
+	s08
+	stm8
+	pdk13 pdk14 pdk15 pdk16
+	mos6502 mos65c02
+"
+IUSE="
+	${SDCC_PORTS}
+	+boehm-gc device-lib doc non-free packihx +sdbinutils sdcdb +sdcpp ucsim
+"
+
+for port in ${SDCC_PORTS}; do
+REQUIRED_USE="${REQUIRED_USE}
+	${port}? ( sdbinutils )
+"
+done
+REQUIRED_USE="${REQUIRED_USE}
+	|| ( ${SDCC_PORTS} )
+"
```

Comment:
`REQUIRED_USE` requires at least one port, and any port requires `sdbinutils`, so this flag is always on.
I removed it and set it to always enabled in a follow-up commit.

This fixed some pkgcheck warnings.